### ISSUE TITLE
Encoding error in jedi_completer with python 2 on Mac

### ISSUE THREAD

**Issue by ptrv** [Valloric/ycmd#419]
I get this error when I try to parse a python file:

```
2016-03-07 11:08:21,374 - INFO - Starting JediHTTP server
2016-03-07 11:08:21,376 - INFO - using port 49625
2016-03-07 11:08:21,377 - ERROR - unknown encoding: 
Traceback (most recent call last):
  File "/Users/peter/src/ycmd/ycmd/../ycmd/server_state.py", line 94, in FiletypeCompletionAvailable
    self.GetFiletypeCompleter( filetypes )
  File "/Users/peter/src/ycmd/ycmd/../ycmd/server_state.py", line 82, in GetFiletypeCompleter
    for filetype in current_filetypes ]
  File "/Users/peter/src/ycmd/ycmd/../ycmd/server_state.py", line 71, in _GetFiletypeCompleterForFiletype
    completer = module.GetCompleter( self._user_options )
  File "/Users/peter/src/ycmd/ycmd/completers/python/hook.py", line 30, in GetCompleter
    return JediCompleter( user_options )
  File "/Users/peter/src/ycmd/ycmd/../ycmd/completers/python/jedi_completer.py", line 75, in __init__
    self._StartServer()
  File "/Users/peter/src/ycmd/ycmd/../ycmd/completers/python/jedi_completer.py", line 172, in _StartServer
    with open( self._logfile_stderr, 'w' ) as logerr:
LookupError: unknown encoding: 
```

**Comment by Valloric** [comment]
Bah, another `open()` call we missed. Thanks for reporting it.

**Comment by micbou** [comment]
I find it strange that you are the only one to report this kind of issue. Just saying but there is probably something wrong with your configuration (maybe your locale?)

**Comment by Valloric** [comment]
Yeah I too find it strange since it works on Travis just fine and passes our tests, but it's actually good because we shouldn't be using the Python implicit encoding. We should specify utf8 always.

**Comment by ptrv** [comment]
I find it also strange, because on my Linux system everything works either with python 2 or python 3. I have the issues only on my Mac at work. I have to investigate what in my configuration leads to this. Still it should be more robust.

**Comment by Valloric** [comment]
> I have to investigate what in my configuration leads to this.

Please don't; you're machine is the only test vector we have for this. :D

**Comment by causlayer** [comment]
Same error message:
2016-03-08 15:24:09,711 - INFO - Received filetype completion available request
2016-03-08 15:24:09,754 - INFO - Starting JediHTTP server
2016-03-08 15:24:09,756 - INFO - using port 54383
2016-03-08 15:24:09,766 - ERROR - unknown encoding: 
Traceback (most recent call last):
  File "/Users/apple/Documents/apple/project/k-vim/bundle/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 94, in FiletypeCompletionAvailable
    self.GetFiletypeCompleter( filetypes )
  File "/Users/apple/Documents/apple/project/k-vim/bundle/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 82, in GetFiletypeCompleter
    for filetype in current_filetypes ]
  File "/Users/apple/Documents/apple/project/k-vim/bundle/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/server_state.py", line 71, in _GetFiletypeCompleterForFiletype
    completer = module.GetCompleter( self._user_options )
  File "/Users/apple/Documents/apple/project/k-vim/bundle/YouCompleteMe/third_party/ycmd/ycmd/completers/python/hook.py", line 30, in GetCompleter
    return JediCompleter( user_options )
  File "/Users/apple/Documents/apple/project/k-vim/bundle/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/completers/python/jedi_completer.py", line 75, in __init__
    self._StartServer()
  File "/Users/apple/Documents/apple/project/k-vim/bundle/YouCompleteMe/third_party/ycmd/ycmd/../ycmd/completers/python/jedi_completer.py", line 172, in _StartServer
    with open( self._logfile_stderr, 'w' ) as logerr:
LookupError: unknown encoding: 

max os x 10.11.3
I get this error when I run :YcmRestartServer 
source file:/private/var/folders/gx/vdqd0d_15ms4ys0wf1cy_9vr0000gn/T/ycm_temp/server_54327_stderr.log

**Comment by ptrv** [comment]
I can confirm that PR #420 fixes this issue on my Mac at work

**Comment by causlayer** [comment]
Thanks @ptrv

**Comment by micbou** [comment]
Fixed by PR #420.
